src/services/FormValueManipulator.ts

```diff
@@ -1,45 +1,20 @@
 import { ComponentFormType } from '../typings/index';
-import EMPTY_VIEW_COMPONENTS from '../constants/emptyViewComponents';
 import { Component } from '../typings/schema';
+import { NormalizedComponent } from '../typings/index';
 
-const renderName = (typename: string, idx: number, field?: string): string => (
+export const renderName = (typename: string, idx: number, field?: string): string => (
     `${idx}_${typename}${field ? '_' + field : ''}`
 );
 
-const setComponentInfo = (item: any, idx: number) => {
-    const newItem = { ...item };
-
-    newItem.componentInfo = {
-        name: renderName(item.type, idx),
-        index: idx
-    };
-
-    return newItem;
+export const addExtraInfo = (item: NormalizedComponent): ComponentFormType => ({
```

Comment:
This can go, you've now got a fancy normalizing reducer to handle this stuff for you